pkg/types/validation/installconfig.go

```diff
@@ -122,6 +123,13 @@ func ValidateInstallConfig(c *types.InstallConfig, usingAgentMethod bool) field.
 	allErrs = append(allErrs, validatePlatform(&c.Platform, usingAgentMethod, field.NewPath("platform"), c.Networking, c)...)
 	if c.ControlPlane != nil {
 		allErrs = append(allErrs, validateControlPlane(&c.Platform, c.ControlPlane, field.NewPath("controlPlane"))...)
+		if len(c.ControlPlane.Fencing.Credentials) > 0 {
+			if c.EnabledFeatureGates().Enabled(features.FeatureGateDualReplica) {
+				allErrs = append(allErrs, validateFencingCredentials(c)...)
+			} else {
+				allErrs = append(allErrs, field.Forbidden(field.NewPath("controlPlane").Child("fencing").Child("credentials"), fmt.Sprintf("%s feature must be enabled in order to use Two Node Fencing (TNF) cluster deployment", features.FeatureGateDualReplica)))
```

Comment:
And I would move the feature gate check to be consistent with the other gated features:

https://github.com/openshift/installer/blob/513176f475eb1c8e840888c86e53a72a884bf6be/pkg/types/validation/installconfig.go#L1454-L1465

You would probably want to add a case where c.ControlPlane != nil